### PR TITLE
implement min/max length rules

### PR DESCRIPTION
**Pull request by zargot**
This PR implements the min/max-length rules. See commit messages for more details.

### CODE REVIEW

File: src/odm_validation/validation.py [hunk 1]
```diff
@@ -57,35 +69,34 @@ def _get_latest_odm_version() -> str:
 _KEY_RULES = {r.key: r for r in ruleset}
 
 
-def _rule_name(rule_id):
-    return rule_id.replace('_', ' ').capitalize()
+def _get_rule_name(rule: Rule):
```
Comment (yulric) [inline src/odm_validation/validation.py:72]: I would rename to `_prettify_rule_name`

File: src/odm_validation/rule_primitives.py [hunk 2]
```diff
@@ -138,19 +138,21 @@ def set_attr_schema(table_schema, data, table, attr, rule_id, odm_key,
     deep_update(attr_schema, table_schema[table_id1]['schema']['schema'])
 
 
-def gen_simple_schema(data: pt.PartData, ver, rule_id, odm_key, cerb_key):
+def gen_simple_schema(data: pt.PartData, ver, rule_id, odm_key, cerb_key,
```
Comment (yulric) [inline src/odm_validation/rule_primitives.py:141]: Can you add type definitions and documentation for each of these parameters?

File: assets/validation-rules/greater-than-max-length/error-report.json [hunk 3]
```diff
@@ -9,14 +9,16 @@
                 "contactID": "1",
                 "phone": "123-456-111"
             },
+            "invalidValue": "123-456-111",
```
Comment (yulric) [inline assets/validation-rules/greater-than-max-length/error-report.json:12]: Add this field to both specs?

File: assets/validation-rules/less-than-min-length/schema-v1.yml [hunk 4]
```diff
@@ -6,17 +6,18 @@ schema:
       type: dict
       schema:
         phoneNumber:
-          minLength: 10
+          minlength: 10
```
Comment (yulric) [inline assets/validation-rules/less-than-min-length/schema-v1.yml:9]: Update line 69 in both specs? minLength -> minlength and maxLength to maxlength

**Review by yulric — CHANGES_REQUESTED** [review]

File: src/odm_validation/validation.py (hunk 1 above)
Comment (zargot) [reply src/odm_validation/validation.py]: done

File: src/odm_validation/rule_primitives.py (hunk 2 above)
Comment (zargot) [reply src/odm_validation/rule_primitives.py]: done

File: assets/validation-rules/greater-than-max-length/error-report.json (hunk 3 above)
Comment (zargot) [reply assets/validation-rules/greater-than-max-length/error-report.json:12]: done

File: assets/validation-rules/less-than-min-length/schema-v1.yml (hunk 4 above)
Comment (zargot) [reply assets/validation-rules/less-than-min-length/schema-v1.yml:9]: done

**Review by yulric — APPROVED** [review]

**Pull request merged**